core/server/common/src/main/java/alluxio/master/journal/Journal.java

```diff
@@ -11,92 +11,117 @@
 
 package alluxio.master.journal;
 
-import alluxio.AlluxioURI;
+import alluxio.master.journal.ufs.UfsJournal;
+import alluxio.master.journal.ufs.UfsMutableJournal;
+import alluxio.util.URIUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * This class encapsulates the journal for a master. The journal is made up of 2 components:
- * - The checkpoint: the full state of the master
- * - The entries: incremental entries to apply to the checkpoint.
- *
- * To construct the full state of the master, all the entries must be applied to the checkpoint in
- * order. The entry file most recently being written to is in the base journal folder, where the
- * completed entry files are in the "completed/" sub-directory.
+ * The read-only journal. It prevents access to a {@link JournalWriter}.
  */
 @ThreadSafe
-public abstract class Journal {
-  /** The log number for the first completed log file. */
-  public static final long FIRST_COMPLETED_LOG_NUMBER = 1L;
-  /** The directory for completed log files, relative to the base journal directory. */
-  private static final String COMPLETED_DIRECTORY = "completed/";
-  /** The file extension for the current log file. */
-  private static final String CURRENT_LOG_EXTENSION = ".out";
-  /** The filename of the checkpoint file. */
-  private static final String CHECKPOINT_FILENAME = "checkpoint.data";
-  /** The base of the entry log filenames, without the file extension. */
-  private static final String ENTRY_LOG_FILENAME_BASE = "log";
-  /** The directory where this journal is stored. */
-  private final String mDirectory;
-  /** The formatter for this journal. */
-  private final JournalFormatter mJournalFormatter;
+public interface Journal {
 
   /**
-   * Creates a new instance of {@link Journal}.
-   *
-   * @param directory the base directory for this journal
+   * A factory which creates read-write journals.
```

Comment:
this can also create read-only journals right? Also, use "mutable" instead of "read-write" to be consistent?